lib/AST/ProtocolConformance.cpp

```diff
@@ -241,7 +241,7 @@ void NormalProtocolConformance::setTypeWitness(
   assert(getProtocol() == cast<ProtocolDecl>(assocType->getDeclContext()) &&
          "associated type in wrong protocol");
   assert(TypeWitnesses.count(assocType) == 0 && "Type witness already known");
-  assert(!isComplete() && "Conformance already complete?");
+  assert((!isComplete() || isInvalid()) && "Conformance already complete?");
```

Comment:
You could say "Conformance already complete or invalid" but then one wouldn't know which condition actually failed, so maybe separating into 2 assertions would be better?